src/types/commits.ts

```diff
@@ -24,3 +26,8 @@ export enum CommitRequesterTypes {
   TriggerRequester = "trigger_request",
   GitTagRequester = "git_tag_request",
 }
+
+export type Commits = MainlineCommitsQuery["mainlineCommits"]["versions"];
+export type Commit = Commits[0];
```

Comment:
[nit] You could probably use the [Unpacked](https://github.com/evergreen-ci/spruce/blob/main/src/types/utils.ts#L2) helper instead